src/core/simulation.jl

```diff
@@ -235,20 +245,44 @@ function _build_stages!(sim::Simulation, verbose::Bool = true; kwargs...)
     return
 end
 
+function _build_stage_paths!(sim::Simulation, verbose::Bool = true; kwargs...)
+    system_to_file = get(kwargs, :system_to_file, true)
+    for (stage_number, stage_name) in sim.sequence.order
+        stage = get(sim.stages, stage_name, nothing)
+        stage_path = joinpath(sim.internal.models_dir, "stage_$(stage_name)_model")
+        mkpath(stage_path)
+        _write_psi_container(stage.internal.psi_container,
+                            joinpath(stage_path, "$(stage_name)_optimization_model.json"))
+        system_to_file && PSY.to_json(stage.sys, joinpath(stage_path , "$(stage_name)_sys_data.json"))
+    end
+end
+
+function _check_folder(folder::String)
+    !isdir(folder) && throw(IS.ConflictingInputsError("Specified folder is not valid"))
+    try
+        mkdir(joinpath(folder, "fake"))
+        rm(joinpath(folder, "fake"))
+    catch e
+        throw(IS.ConflictingInputsError("Specified folder does not have write access [$e]"))
+    end
+end
+
 function build!(sim::Simulation; verbose::Bool = false, kwargs...)
```

Comment:
This function needs a docstring since it's a key user interface. It can wait for a seperate PR if you prefer.